src/common/selftest/selftest_esp_update.cpp

```diff
@@ -297,11 +297,14 @@ bool EspCredentials::make_file() {
         return false;
     }
 
-    fputs(file_str, file.get());
+    bool failed = fputs(file_str, file.get()) == EOF;
+    if (failed) {
```

Comment:
I'd recommend a more general error check ... something like
`if( fputs(...) < 0 ) ... log error`

Reply:
https://www.cplusplus.com/reference/cstdio/fputs/

Return Value
On success, a non-negative value is returned.
On error, the function returns EOF.